server/src/main/java/org/opensearch/tracing/DefaultTracer.java

```diff
@@ -0,0 +1,229 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.tracing;
+
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.context.Context;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.common.util.concurrent.ThreadContext;
+import org.opensearch.threadpool.ThreadPool;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.Optional;
+
+/**
+ *
+ * The default tracer implementation. This class implements the basic logic for span lifecycle and its state management.
+ * It also handles tracing context propagation between spans.
+ *
+ * It internally uses OpenTelemetry tracer.
+ *
+ */
+public class DefaultTracer implements Tracer {
```

Comment:
Agree on extracting Otel based DefaultTracer in a separate module. Have made the changes in new revision.

Considering Opentelemetry is already standardised for tracing, we currently do not  plan to make it plugin/extension. However since we have have extracted the interfaces in core and and all Otel specific implementation are extracted in a module, we can move it to plugin/extension if needed.